types/core.d.ts

```diff
@@ -65,7 +65,7 @@ type StrapiEventsPublishFlow = 'publish' | 'unpublish';
 export type StrapiService = any;
 export type StrapiController = any;
 export type StrapiMiddleware = Object;
-export type StrapiContentType<T extends StrapiContentTypeSchema> = T | Object;
+export type StrapiContentType<T extends StrapiContentTypeSchema> = T;
```

Comment:
Shouldn't it extend StrapiContentTypeFullSchema?

Reply:
My rationale is that this type will be used more often (declaring a model) than full schema (with stuff from strapi added at runtime). Forcing people to `Omit` or monkey patching might be annoying.
I'm not 100% sure in my viewpoint.